devnets/startNetwork.ts

```diff
@@ -361,3 +346,8 @@ function addXcmHrmpChannels(
     }
   }
 }
+
+async function getLibp2pPeerId(privateKey: Uint8Array) {
+  const publicKey = await ed25519.getPublicKeyAsync(privateKey)
+  return base58.encode(new Uint8Array([0, 36, 8, 1, 18, 32, ...publicKey]))
```

Comment:
Libp2p encodes keys in a [protobuf](https://github.com/libp2p/go-libp2p/blob/master/core/crypto/pb/crypto.proto) containing a key *type* and the encoded key (where the encoding depends on the type).
Peer IDs are derived by hashing the encoded public key with [multihash](https://github.com/multiformats/multihash). Keys that serialize to more than 42 bytes must be hashed using sha256 multihash, keys that serialize to at most 42 bytes must be hashed using the "identity" multihash codec.

Therefore, the `peerId` is derived as follows
```
peerId = multihash(protobuf.encode(ed25519PrivateKey))
```
The first 6 bytes of `[0, 36, 8, 1, 18, 32, ...publicKey]` are constant for a 32-byte ed25519 private key and it is derived from
```
// for [0, 36]
> import { identity } from 'npm:multiformats/hashes/identity'
undefined
> identity.digest(new Uint8Array(36))
Digest {
  code: 0,
  size: 36,
  digest: Uint8Array(36) [
    0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0, 0, 0, 0, 0, 0,
    0, 0, 0, 0, 0, 0, 0, 0, 0
  ],
  bytes: Uint8Array(38) [
    0, 36, 0, 0, 0, 0, 0, 0, 0, 0,
    0,  0, 0, 0, 0, 0, 0, 0, 0, 0,
    0,  0, 0, 0, 0, 0, 0, 0, 0, 0,
    0,  0, 0, 0, 0, 0, 0, 0
  ]
}

// for [8, 1, 18, 32]
> import {keysPBM} from "npm:@libp2p/crypto/keys"
undefined
> keysPBM.PublicKey.encode({Type:"Ed25519",Data:new Uint8Array(32)})
Buffer(36) [
  8, 1, 18, 32, 0, 0, 0, 0, 0,
  0, 0,  0,  0, 0, 0, 0, 0, 0,
  0, 0,  0,  0, 0, 0, 0, 0, 0,
  0, 0,  0,  0, 0, 0, 0, 0, 0
]
```